tests/cases/conformance/types/typeRelationships/typeInference/contextualSignatureInstantiation.ts

```diff
@@ -19,7 +19,9 @@ var b: number | string;
 var b = foo(g);            // Error, number and string are disjoint types
 var b = bar(1, "one", g);  // Error, number and string are disjoint types
 var b = bar("one", 1, g);  // Error, number and string are disjoint types
-var b = baz(b, b, g);      // Should be number | string
+
+var b2: number | string;
+var b2 = baz(b2, b2, g);      // Should be number | string

```

Comment:
why the change here? it's because now the previous calls are able to change the assigned type to `b` (by not ignoring generic functions we infer string/number instead of unknown). So the contextual return type here changed from `string | number` to `string` and that introduced the error. But that never was the intention of this test case anyway - so I just separated this to decouple this from CFA type changes